### PR TITLE
Typescript Upgrade

### PR DESCRIPTION
**Pull request by rawphp**
Upgraded to use TypeScript and added tests

### CODE REVIEW

File: src/lib/getVersion.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Get version from config.
+ *
+ * @param {String} version configuration object
+ *
+ * @returns {String} version string
+ */
+export default function getVersion(version: string): string {
+  if (version === 'latest') {
+    return Moment().unix().toString();
```
Comment (kivlor) [inline src/lib/getVersion.ts]: if this is the only place you're using moment I'd suggest ditching for their one liner - https://github.com/moment/moment/blob/497f918515ae6ab900f008f19523b1e4ae5e2450/src/lib/moment/to-type.js#L6

Comment (rawphp) [reply src/lib/getVersion.ts]: removed Moment dependency

File: src/lib/getVersion.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Get version from config.
+ *
+ * @param {String} version configuration object
+ *
+ * @returns {String} version string
+ */
+export default function getVersion(version: string): string {
+  if (version === 'latest') {
+    return Math.floor(this.valueOf() / 1000).toString();
```
Comment (kivlor) [inline src/lib/getVersion.ts]: `this` should be `new Date()` :wink:

Comment (kivlor) [reply src/lib/getVersion.ts]: `Math.floor(new Date().valueOf()/1000)`

Comment (rawphp) [reply src/lib/getVersion.ts]: yep my bad... fixed

**Pull request merged**